packages/components/src/components/optgroup/optgroup.component.ts

```diff
@@ -0,0 +1,74 @@
+import type { PropertyValues } from 'lit';
+import { CSSResult, html, nothing } from 'lit';
+import { property, queryAssignedElements } from 'lit/decorators.js';
+import { Component } from '../../models';
+import { DataAriaLabelMixin } from '../../utils/mixins/DataAriaLabelMixin';
+import { DisabledMixin } from '../../utils/mixins/DisabledMixin';
+import { TYPE, VALID_TEXT_TAGS } from '../text/text.constants';
+import { HEADER_ID } from './optgroup.constants';
+import styles from './optgroup.styles';
+
+/**
+ * optgroup component, which creates a grouping of options within a listbox element.
+ *
+ * @dependency mdc-text
+ *
+ * @tagname mdc-optgroup
+ *
+ * @slot default - This is a default slot for mdc-option elements.
+ *
+ * @cssproperty --mdc-optgroup-disabled-color - Allows customization of the disabled option color.
+ */
+class Optgroup extends DataAriaLabelMixin(DisabledMixin(Component)) {
+  /**
+   * The header text to be displayed on the top of the options list.
+   */
+  @property({ type: String, reflect: true }) label?: string;
+
+  /** @internal */
+  @queryAssignedElements({ selector: 'mdc-option' })
+  options!: Array<HTMLElement>;
+
+  private setDisabledForAllOptions(): void {
+    this.options.forEach((element) => {
+      if (this.disabled) {
+        element.setAttribute('disabled', 'true');
```

Comment:
```suggestion
        element.setAttribute('disabled', '');
```
Since its a boolean value, we need to set it in this manner.